### PR TITLE
[JSHINT] enabled on project

### PR DESCRIPTION
**Pull request by ainthek**
Let's start with default {} .jshintrc and ignore tests for start.

to lint:

	npm run lint

coding style #1359

### CODE REVIEW

**Comment by ainthek** [comment]
I have added devDependency to JSHint but build failed: can anyone translate from "travis" to "human" for me? thanx...
https://travis-ci.org/mochajs/mocha/jobs/49659011

**Comment by dasilvacontin** [comment]
You need to make it install a newer `npm` version. We had the same issue over at mustache.js: https://github.com/janl/mustache.js/pull/421

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"laxcomma":true,
+	"laxbreak":true	
+}
```
Comment (boneskull) [inline .jshintrc]: We'll need to define a proper `.jshintrc`, but this is fine for now.

File: package.json [hunk 2]
```diff
@@ -39,7 +39,8 @@
     "node": ">= 0.8.x"
   },
   "scripts": {
-    "test": "make test-all"
+    "test": "make test-all",
+    "lint": "jshint _mocha.js"
```
Comment (boneskull) [inline package.json]: Was the intent to test only one file?

**Comment by boneskull** [comment]
We need to make this part of the `npm test` suite, not a separate script.

**Comment by boneskull** [comment]
@ainthek To that end, you'll need to modify `Makefile` and have it run `jshint` before it executes any tests.

**Comment by boneskull** [comment]
Thanks for this!

**Comment by ainthek** [comment]
Chris, actually I can make this directly in package.json, and script:test can run both lint and the tests.
However I like this to be separated, test is test lint is lint. If you consider lint part of tests please do whatever you want. See: https://github.com/ainthek/node-boilerplate/blob/master/package.json

Decision is really up to you falks....

**Comment by boneskull** [comment]
@ainthek What I'd really like is this to happen:

- new target `lint` in `Makefile`:

  ```Makefile
  lint: 
      npm run-script lint
  ```
- `test` target changed to:

  ```Makefile
  test: lint test-all
  ```
- in `scripts` property of `package.json`:
  
  ```json
  { 
    "test": "make test"
  }
  ```

This gives us the ability to run `jshint` via `make` or `npm run-script`, runs all tests by default, and runs the lint before the tests.

If you want to skip the linting and just run the tests, `make test-all` will be sufficient.

**Comment by ainthek** [comment]
chris: you have already wrote more lines of the git comments then the actual lines of source code needed to implement this. Why ? I do not get this. 

PR is proposal. You have minimal work overhead to adjust it and commit in "adjusted work".
I'm just curious. Why not just making this happen ?

**Comment by boneskull** [comment]
@ainthek 

Project maintainers ask submitters of PRs to change the code for a couple reasons:

- Contrary to what you may think, it will take longer for me to apply the change myself than to write this comment.
- It is considered poor etiquette to take a PR and put your name on it, which would be what I would be doing if I just made the changes myself.

So, if you are not willing to make the changes, we'll close the issue, make a note in #1359, and someone will get to it eventually.  However, I'd like to merge your changes in (with modifications), so I'm asking if you would please change the code and squash your commits.

Thanks!

**Comment by dasilvacontin** [comment]
We actually have an issue discussing that we want to enforce the code style, because it is currently a mess. And it is both easier and faster to make `lint` be a part of the `test` routine rather than having to review the PR, ask for code style changes, wait for style changes, review PR again, etc.

**Comment by ainthek** [comment]
Will shuash and send again.

**Comment by dasilvacontin** [comment]
@ainthek ?

**Comment by ainthek** [comment]
I will work on more complete fix and then make pull request.
Also will try to run travis on my repo first to avoid "bad pull requests".

Will make PR then.

**Pull request closed without merging**